romancal/source_catalog/psf.py

```diff
@@ -134,6 +146,184 @@ def get_gridded_psf_model(psf_ref_model):
+    model contains some potentially different amount of jitter.  This
+    function needs to deconvolve the jitter in the reference model
+    and then reconvolve with the jitter in the image, to obtain a model
+    for the jitter in this image.
+
+    This procedure is made more challenging by the fact that the jitter is
+    small.  Typical values of the jitter are 8 mas compared to a typical
+    PSF FWHM of ~100 mas.  The pixel scale in the reference files is 4x
+    oversampled relative to the native scale of 0.11", and so is ~28 mas---
+    much larger than the 8 mas jitter scale, so naive direct convolution
+    will not work.
+
+    We use the approach of Lang (2020) to do the convolutions in the Fourier
+    domain to avoid this problem.  The key element there is to analytically
+    evaluate the jitter kernel in Fourier space rather than to evaluate
+    it in real space and then transform it.  If the PSF is well sampled,
+    all fourier components corresponding to higher frequencies than the Nyquist
+    frequency will be exactly zero.  Accordingly, if we evaluate the jitter
+    kernel in the FFT domain analytically only over the frequencies where the
+    PSF FFT is non-zero, we can multiply the two FFTs and transform back
+    to real space without needing to think about the fact that the jitter
+    kernel is undersampled in real space.
+
+    Parameters
+    ----------
+    psf_ref_model : datamodels.EpsfRefModel
+        ePSF reference models from CRDS
+    image_model : datamodels.ImageModel
+        image for which a PSF model is desired
+    pixel_scale : float
+        pixel scale of the image.  Because the jitter kernel is very small,
+        this does not need to be terribly accurate; it's used to transform
+        the real-space jitter kernel specification into the pixel-space
+        kernels that are convolved with the PSF stamps.
+
+    Returns
+    -------
+    convolved_stamps : np.ndarray
+        PSF reference file stamps after convolution with jitter kernel
+    """
+
+    ref_jitter_params = _get_jitter_params(psf_ref_model.meta)
+    image_jitter_params = _get_jitter_params(
+        getattr(image_model.meta, "guide_star", dict())
+    )
+    oversample = psf_ref_model.meta.oversample
+    shape = psf_ref_model.psf.shape[-2:]
+    jit_ref_fft = _evaluate_gaussian_fft(
+        ref_jitter_params, shape, pixel_scale / oversample
+    )
+    jit_img_fft = _evaluate_gaussian_fft(
+        image_jitter_params, shape, pixel_scale / oversample
+    )
+    jitter_fft = jit_img_fft / jit_ref_fft
+    stamp_fft = np.fft.rfft2(psf_ref_model.psf, axes=(-2, -1))
+    convolved = np.fft.irfft2(stamp_fft * jitter_fft[None, ...], axes=(-2, -1), s=shape)
+    log.info(
+        "Performed jitter convolution.  "
+        + ("Image kernel: %.1f %.1f %.1f" % tuple(image_jitter_params.values()))
+        + ("Reference kernel: %.1f %.1f %.1f" % tuple(ref_jitter_params.values()))
+    )
```

Comment:
```suggestion
    log.info(f"Performed jitter convolution.  Image kernel: {tuple(image_jitter_params.values()))} Reference kernel: tuple(ref_jitter_params.values()))")
```
Hopefully this will appease ruff (after pre-commit reformats the long line).